### DISCUSSION TITLE
Change font size

### DISCUSSION THREAD

**Discussion by DanielRodrigues1** [olikraus/u8g2#1744] [Q&A] [answered]
Hi, I have been testing the U8g2 library, i only want to know if theres by any chance, a way to change de font or text size.  In the original Adafruit.GFX library i have `setTextSize();` function, wich not exactly change the font size, but it's one way to manipulate the text size.

is there something similar in the U8g2 library?

regardless, thanks.

**Comment by olikraus** [comment]
The Adafruit setTextSize() will just extend each pixel, which will be fast, but will not look good:
![https://cdn-learn.adafruit.com/assets/assets/000/001/277/medium640/lcds___displays_text.jpg?1396770564](https://cdn-learn.adafruit.com/assets/assets/000/001/277/medium640/lcds___displays_text.jpg?1396770564)

U8g2 does not support such scaling of the pixels.
Instead the idea is to use the same font type in multiple sizes, for example [helvB08](https://github.com/olikraus/u8g2/wiki/fntgrpadobex11#helvr08) and [helvB12](https://github.com/olikraus/u8g2/wiki/fntgrpadobex11#helvr12)  

![https://github.com/olikraus/u8g2/wiki/fntpic/u8g2_font_helvB08_tr.png](https://github.com/olikraus/u8g2/wiki/fntpic/u8g2_font_helvB08_tr.png)

![https://github.com/olikraus/u8g2/wiki/fntpic/u8g2_font_helvB12_tr.png](https://github.com/olikraus/u8g2/wiki/fntpic/u8g2_font_helvB12_tr.png)

Note that most embedded controllers are not fast enough and do not have enough memory to do a proper scaling of fonts.

**Reply by DanielRodrigues1** [reply]
Got it, so if i want a font with only one size available, i have to resize the font by hand?

one more quick question, can i change the background color of the font? by the default is black right? but i need to be red to match the tft background, is it possible?

**Comment by olikraus** [comment]
> so if i want a font with only one size available, i have to resize the font by hand?

Correct, with a .ttf font for example you can generate the .bdf file out of it with a different size (see the FAQ). 

 > can i change the background color of the font?

U8g2 is a monochrome graphics library. It only deals with enabled or disabled pixel values. Whether an enabled pixel appears white, blue, pink or black is defined by your display. Same is true for a disabled pixel value. Whatever color is visible if the pixel value is zero depends on your display. Usually, if you need red background color, you need to get a display with red LED backlight (see here for example: https://www.lcd-module.com/produkte/dog.html).

**Reply by DanielRodrigues1** [reply]
Im using a ili 9431 but when i try to chang the background color of the display with the `fill` function its like this 

![WhatsApp Image 2022-01-07 at 20 38 25](https://user-images.githubusercontent.com/72764143/148691745-519b2beb-d652-461b-af89-38159a36061c.jpeg)

im trying to get rid of this black squares under the font, or make them blue to match the background, is it possible in the u8g2 library?

**Comment by olikraus** [comment]
I am confused. U8g2 does not support ili 9431.
If you use u8g2_for_adafruit_gfx then you need to set font mode 1 to turn of the black boxes: `u8g2_for_adafruit_gfx.setFontMode(1)`. This is also suggested in the example code here: https://github.com/olikraus/U8g2_for_Adafruit_GFX

**Reply by DanielRodrigues1** [reply]
Yes i am using u8g2_for_adafruit_gfx, sorry for not have specified correctly.  Ive tried to use `u8g2_for_adafruit_gfx.setFontMode(1)` but when a number is updated in the display , it just overwrite himself making a mess, maybe i just have to clean the display everytime something changes?

**Comment by olikraus** [comment, marked as answer]
>  it just overwrite himself making a mess, 

ah ok, in this case you must use a font with a "_hr" or "_mr" postfix (h or m instead of the t).

And of course you need to apply a proper background color: u8g2_for_adafruit_gfx.setBackgroundColor(uint16_t color)

Here are some fonts which will work better in your case:
```
u8g2_font_micro_mr
u8g2_font_4x6_mr
u8g2_font_5x7_mr
u8g2_font_5x8_mr
u8g2_font_6x10_mr
u8g2_font_6x12_mr
u8g2_font_6x13_mr
u8g2_font_6x13B_mr
u8g2_font_6x13O_mr
u8g2_font_7x13_mr
u8g2_font_7x13B_mr
u8g2_font_7x13O_mr
u8g2_font_7x14_mr
u8g2_font_7x14B_mr
u8g2_font_8x13_mr
u8g2_font_8x13B_mr
u8g2_font_8x13O_mr
u8g2_font_9x15_mr
u8g2_font_9x15B_mr
u8g2_font_9x18_mr
u8g2_font_9x18B_mr
u8g2_font_10x20_mr
u8g2_font_tom_thumb_4x6_mr
u8g2_font_t0_11_mr
u8g2_font_t0_11b_mr
u8g2_font_t0_12_mr
u8g2_font_t0_12b_mr
u8g2_font_t0_13_mr
u8g2_font_t0_13b_mr
u8g2_font_t0_14_mr
u8g2_font_t0_14b_mr
u8g2_font_t0_15_mr
u8g2_font_t0_15b_mr
u8g2_font_t0_16_mr
u8g2_font_t0_16b_mr
u8g2_font_t0_17_mr
u8g2_font_t0_17b_mr
u8g2_font_t0_18_mr
u8g2_font_t0_18b_mr
u8g2_font_t0_22_mr
u8g2_font_t0_22b_mr
u8g2_font_profont10_mr
u8g2_font_profont11_mr
u8g2_font_profont12_mr
u8g2_font_profont15_mr
u8g2_font_profont17_mr
u8g2_font_profont22_mr
u8g2_font_profont29_mr
u8g2_font_tenstamps_mr
u8g2_font_VCR_OSD_mr
u8g2_font_crox1cb_mr
u8g2_font_crox1c_mr
u8g2_font_crox2cb_mr
u8g2_font_crox2c_mr
u8g2_font_crox3cb_mr
u8g2_font_crox3c_mr
u8g2_font_cu12_mr
u8g2_font_chargen_92_mr
u8g2_font_inr16_mr
u8g2_font_inr19_mr
u8g2_font_inr21_mr
u8g2_font_inr24_mr
u8g2_font_inr27_mr
u8g2_font_inr30_mr
u8g2_font_inr33_mr
u8g2_font_inr38_mr
u8g2_font_inr42_mr
u8g2_font_inr46_mr
u8g2_font_inr49_mr
u8g2_font_inr53_mr
u8g2_font_inb16_mr
u8g2_font_inb19_mr
u8g2_font_inb21_mr
u8g2_font_inb24_mr
u8g2_font_inb27_mr
u8g2_font_inb30_mr
u8g2_font_inb33_mr
u8g2_font_inb38_mr
u8g2_font_inb42_mr
u8g2_font_inb46_mr
u8g2_font_inb49_mr
u8g2_font_inb53_mr
u8g2_font_pxplusibmvga9_mr
u8g2_font_pxplusibmvga8_mr
u8g2_font_px437wyse700a_mr
u8g2_font_px437wyse700b_mr
```

As an alternative solution: You could clear the previous data with a rectangle and then write the new text.

**Reply by DanielRodrigues1** [reply]
I think this should do it, tomorrow i will run some test and i will tell you the results

thanks a lot olliver.

**Reply by DanielRodrigues1** [reply]
`u8g2_for_adafruit_gfx.setBackgroundColor(uint16_t color) ` was the command that i was looking for, work's fine now, thanks oliver

**Reply by olikraus** [reply]
ah, ok :-)
